openmetadata-ui/src/main/resources/ui/src/pages/TeamsAndUsersPage/TeamsAndUsersPage.component.tsx

```diff
@@ -646,83 +691,100 @@ const TeamsAndUsersPage = () => {
   };
 
   useEffect(() => {
-    if (teamAndUser) {
-      if (Object.values(UserType).includes(teamAndUser as UserType)) {
-        setIsTeamVisible(false);
-        setactiveUserTab(teamAndUser as UserType);
-        setCurrentTeam(undefined);
+    if (!isDataLoading) {
+      if (teamAndUser) {
+        if (Object.values(UserType).includes(teamAndUser as UserType)) {
+          setIsTeamVisible(false);
+          setactiveUserTab(teamAndUser as UserType);
+          setCurrentTeam({} as Team);
+          setCurrentUserPage(INITIAL_PAGIN_VALUE);
+          setAllTabList(teamAndUser as UserType);
+          setUserPaging({} as Paging);
+        } else {
+          fetchCurrentTeam(teamAndUser);
+        }
       } else {
-        fetchCurrentTeam(teamAndUser);
+        setactiveUserTab(undefined);
```

Comment:
please avoid assigning undefined directly to a variable or state . alternate approach  etactiveUserTab({});